File: backend/app/models/command.py
from datetime import datetime
from sqlalchemy import DateTime, Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from app.db.session import Base


class Command(Base):
    __tablename__ = "commands"

    command_id: Mapped[str] = mapped_column(String(64), primary_key=True, index=True)
    drone_id: Mapped[str] = mapped_column(String(64), index=True)
    operator_id: Mapped[str] = mapped_column(String(64))
    command_type: Mapped[str] = mapped_column(String(64))
    status: Mapped[str] = mapped_column(String(32), default="QUEUED")
    decision: Mapped[str] = mapped_column(String(32))
    reason: Mapped[str] = mapped_column(Text)
    ack_message: Mapped[str | None] = mapped_column(Text, nullable=True)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True))
    acknowledged_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), nullable=True)

File: frontend/src/types/command.ts
export type CommandType = 'READ_STATUS' | 'START_LOGGING' | 'STOP_LOGGING' | 'SIMULATE_RTL' | 'SIMULATE_LAND';
export interface Command { command_id: string; drone_id: string; operator_id: string; command_type: CommandType; status: string; decision: string; reason: string; ack_message?: string; created_at: string; acknowledged_at?: string; }
